libsupport/include/katana/Result.h

```diff
@@ -260,6 +260,40 @@ ResultErrno() {
   return std::error_code(errno, std::system_category());
 }
 
+template <class>
+struct is_result : std::false_type {};
+
+template <class T>
+struct is_result<Result<T>> : std::true_type {};
+
+template <class T>
+std::enable_if_t<!std::is_same<T, void>::value, T&&>
+extract_value(Result<T>&& result) {
+  return std::move(result.value());
+}
+
+inline int
+extract_value(Result<void>&&) {
+  return 0;
+}
+
+#define KATANA_CHECK_NAME(x, y) x##y
+
+#define KATANA_CHECK_IMPL(result_name, expression)                             \
+  ({                                                                           \
+    auto result_name = (expression);                                           \
+    static_assert(                                                             \
+        ::katana::is_result<decltype(result_name)>::value,                     \
+        "KATANA_CHECK requires a katana::Result");                             \
+    if (!result_name) {                                                        \
+      return result_name.error().WithContext("here");                          \
```

Comment:
Why "here"? Also how is this picking up the stack information? Doesn't there need to be a macro that adds it?

Reply:
I don't understand how WithContext works, but it picks up stack information itself.
I also seem to have lost a commit somewhere, "here" is certainly not final and is a parameter to the (missing) `KATANA_CHECK_CONTEXT`